internal/ethapi/api.go

```diff
@@ -1165,13 +1165,28 @@ func doCall(ctx context.Context, b Backend, args TransactionArgs, state *state.S
 	return result, nil
 }
 
+func updateHeaderForPendingBlocks(blockNrOrHash rpc.BlockNumberOrHash, header *types.Header) *types.Header {
+	if blockNrOrHash.BlockNumber != nil &&
+		*blockNrOrHash.BlockNumber == rpc.PendingBlockNumber {
+		headerCopy := *header
+		blkTime := uint64(time.Now().Unix())
+		if blkTime > headerCopy.Time {
+			headerCopy.Time = blkTime
+		}
+		headerCopy.Number = big.NewInt(headerCopy.Number.Int64() + 1)
```

Comment:
Let's do `headerCopy.Number = new(big.Int).Add(headerCopy.Number, common.Big1)`, which is a bit more efficient